spec/support/capybara_actions.rb

```diff
@@ -10,13 +10,19 @@ def within_section(title, &)
     within(:xpath, "//section[contains(header/text(),'#{title}')]", &)
   end
 
+  # NOTE: this is here to ensure all turbo frames are loaded before auditing a11y
+  def visit_and_wait_for_complete_frames(path)
+    visit path
+    # This is admittedly an imperfect strategy.
+    all(:xpath, ".//turbo-frame[@complete='']")
```

Comment:
Can you explain what this is saying?

Reply:
It forces Capybara to try to find all `<turbo-frame>` elements on the page that have already completed. Maybe there's a smarter way to get Capybara to wait until turbo-frames have loaded. If so, I'm all ears.

Reply:
Not questioning it, just trying to understand. Can you unpack how it works?

Reply:
@justinlittman Sure, not a problem. `all` is a Capybara finder method that searches the page for a given selector and will wait `Capybara.default_max_wait_time` seconds for matches. It's arguably a slightly smarter `sleep` (also, arbuably, a slightly dumber `sleep`). I'll add this to the comment in the code.

Reply:
https://rubydoc.info/github/teamcapybara/capybara/master/Capybara/Node/Finders#all-instance_method

Reply:
@justinlittman I've read that doc about five times between yesterday and today and I'm still fuzzy on that. Let me try making it a bit smarter.

Reply:
Pushing up a tweak momentarily. Thanks for taking a look at this.

Reply:
I wonder if the code comments might have links to the relevant docs?  Or otherwise give us a steer when we come back to this in 2 years

Reply:
@ndushay Maybe lost in the shuffle, but ☝🏻 

> **I'll add this to the comment in the code.**